File: CPE-1star/UVA10812-Beat the Spread!.py
#兩數加起來除2和兩數相減除2
while True:
	try:
		n=int(input())
		for i in range(n):
			s,d=map(int,input().split())
			if s<d or (s-d)%2!=0:
				print("impossible")
			else:
				print(d+(s-d)//2,end=" ")
				print((s-d)//2)				
	except:
		break

#2
while True:
	try:
		n=int(input())
		for i in range(n):
			s,d=map(int,input().split())
			if s<d or (s-d)%2!=0:
				print("impossible")
			else:
				print((s+d)//2,end=" ")
				print((s-d)//2)
	except:
		break

File: CPE-1star/UVA10812-Beat the Spread!.cpp
#include<iostream>
using namespace std;
int main(){
    int n;
    while(cin>>n){
        for(int i=0;i<n;i++){
            int a,b;
            cin>>a>>b;
            if(a<b | (a-b)%2!=0)
                cout<<"impossible"<<'\n';
            else{
                cout<<(a+b)/2<<" "<<(a-b)/2<<'\n';
            }
        }
    }
    return 0;
}